controllers/datadogagent/feature/helmcheck/feature.go

```diff
@@ -0,0 +1,156 @@
+		reqComp.ClusterAgent.IsRequired = apiutils.NewBoolPointer(true)
+		reqComp.Agent.IsRequired = apiutils.NewBoolPointer(true)
+
+		f.configMapName = fmt.Sprintf("%s-%s", f.owner.GetName(), apicommon.DefaultHelmCheckConf)
+		f.collectEvents = apiutils.BoolValue(helmCheck.CollectEvents)
+		f.valuesAsTags = helmCheck.ValuesAsTags
+		f.serviceAccountName = v2alpha1.GetClusterAgentServiceAccount(dda)
+
+		if v2alpha1.IsClusterChecksEnabled(dda) && v2alpha1.IsCCREnabled(dda) {
+			f.runInClusterChecksRunner = true
+			f.rbacSuffix = common.ChecksRunnerSuffix
+			f.serviceAccountName = v2alpha1.GetClusterChecksRunnerServiceAccount(dda)
+			reqComp.ClusterChecksRunner.IsRequired = apiutils.NewBoolPointer(true)
+		}
+	}
+
+	return reqComp
+}
+
+// ConfigureV1 use to configure the feature from a v1alpha1.DatadogAgent instance.
+func (f *helmCheckFeature) ConfigureV1(dda *v1alpha1.DatadogAgent) feature.RequiredComponents {
+	return feature.RequiredComponents{}
+}
+
+// ManageDependencies allows a feature to manage its dependencies.
+// Feature's dependencies should be added in the store.
+func (f *helmCheckFeature) ManageDependencies(managers feature.ResourceManagers, components feature.RequiredComponents) error {
+	// Create configMap based on feature flags.
+	cm, err := f.buildHelmCheckConfigMap()
+	if err != nil {
+		return err
+	}
+
+	if err := managers.Store().AddOrUpdate(kubernetes.ConfigMapKind, cm); err != nil {
+		return err
+	}
+
+	// Manage RBAC permission
+	rbacName := getHelmCheckRBACResourceName(f.owner, f.rbacSuffix)
+
+	return managers.RBACManager().AddClusterPolicyRules(f.owner.GetNamespace(), rbacName, f.serviceAccountName, helmCheckRBACPolicyRules)
+}
+
+// ManageClusterAgent allows a feature to configure the ClusterAgent's corev1.PodTemplateSpec
+// It should do nothing if the feature doesn't need to configure it.
+func (f *helmCheckFeature) ManageClusterAgent(managers feature.PodTemplateManagers) error {
+	// Manage Helm check config in configMap
+	var vol corev1.Volume
+	var volMount corev1.VolumeMount
+	// Mount default volumes for configMap
+	vol = volume.GetBasicVolume(f.configMapName, apicommon.HelmCheckConfigVolumeName)
+	volMount = corev1.VolumeMount{
+		Name:      apicommon.HelmCheckConfigVolumeName,
+		MountPath: fmt.Sprintf("%s%s/%s", apicommon.ConfigVolumePath, apicommon.ConfdVolumePath, helmCheckFolderName),
+		ReadOnly:  true,
+	}
+
+	managers.VolumeMount().AddVolumeMountToContainer(&volMount, apicommonv1.ClusterAgentContainerName)
+	managers.Volume().AddVolume(&vol)
+
+	return nil
+}
+
+// ManageMultiProcessNodeAgent allows a feature to configure the multi-process container for Node Agent's corev1.PodTemplateSpec
+// if multi-process container usage is enabled and can be used with the current feature set
+// It should do nothing if the feature doesn't need to configure it.
+func (f *helmCheckFeature) ManageMultiProcessNodeAgent(managers feature.PodTemplateManagers, provider string) error {
```

Comment:
It seems you merged after the mono-container rename PR was merged. 
This one should be delete since we have new one on line 61. Perhaps better to move that one here, closer to `ManageNodeAgent`.